### PR TITLE
Remove timeop

### LINKED ISSUE #19274: (Datetime|Timedelta)Index +/- np.nan
AFAICT this is now the last thing standing between us and getting rid of ops._Op and ops._TimeOp.  Series ops currently case np.nan to pd.NaT whereas Index ops treat it as a float and raise a TypeError.

```
tdi = pd.TimedeltaIndex(['1 Day'])
ser = pd.Series(tdi)

>>> tdi + np.nan
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'TimedeltaIndex' and 'float'

>>> ser + np.nan
0   NaT
dtype: timedelta64[ns]
```

Note also that the behavior of Series[datetime64] has changed since 0.22.0:

0.22.0:
```
dti = pd.date_range('2016-01-01', periods=2)
ser = pd.Series(dti)

>>> dti - np.nan
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for -: 'DatetimeIndex' and 'float'

>>> ser - np.nan
0   NaT
1   NaT
dtype: timedelta64[ns]

# Both raise on addition, assuming that np.nan gets cast to pd.NaT wearing its datetime hat.
```

ATM `ser - np.nan` also raises, since that now delegates to the DatetimeIndex op.

### PR DESCRIPTION
**Pull request by jbrockmendel**
We're finally able to remove `ops._Op` and `ops._TimeOp`.  This PR makes one final compatibility fix:

ATM:
```
tdi = pd.TimedeltaIndex(['1 Day'])
ser = pd.Series(tdi)

>>> tdi + np.nan
TypeError: unsupported operand type(s) for +: 'TimedeltaIndex' and 'float'

>>> ser + np.nan
0   NaT
dtype: timedelta64[ns]
```

Same deal for subtracting np.nan.  See #19274.  This makes the TimedeltaIndex canonical and updates the appropriate Series tests.

- [x] closes #19274
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -6,31 +6,30 @@
...
-                          tslib as libts, algos as libalgos, iNaT)
+                          algos as libalgos)
 
 from pandas import compat
 from pandas.util._decorators import Appender
 
 from pandas.compat import bind_method
 import pandas.core.missing as missing
 
-from pandas.errors import PerformanceWarning, NullFrequencyError
```
Comment (jreback) [inline pandas/core/ops.py]: lgtm. are there any of these PerformanceWarnings uncaught?

Comment (jbrockmendel) [reply pandas/core/ops.py]: Pretty sure we got them all while implementing the Index versions of these ops.

**Comment by jbrockmendel** [comment]
Travis cancelled. Repush or let sit?

**Comment by jreback** [comment]
i restarted

**Comment by jbrockmendel** [comment]
Looks like OSX build got killed again, otherwise OK

**Comment by jreback** [comment]
lgtm. just waiting on travis.

**Comment by jreback** [comment]
thanks.

now would be a good time to split the test_operators into sub-modules :>

**Pull request merged**